terraform/modules/jvs-services/variables.tf

```diff
@@ -0,0 +1,213 @@
+# Copyright 2023 The Authors (see AUTHORS file)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+variable "project_id" {
+  type        = string
+  description = "The GCP project to host the justification verification service."
+}
+
+variable "region" {
+  type        = string
+  default     = "us-central1"
+  description = "The default Google Cloud region to deploy resources in (defaults to 'us-central1')."
+}
+
+variable "service_ingress" {
+  type        = string
+  default     = "all"
+  description = "The ingress settings for user facing services, possible values: all, internal, internal-and-cloud-load-balancing (defaults to 'all')."
+}
+
+variable "jvs_api_service_name" {
+  description = "Name for JVS API service."
+  type        = string
+  default     = "jvs-api"
+}
+
+variable "jvs_api_service_account" {
+  description = "Service account for JVS API service."
+  type        = string
+}
+
+variable "jvs_api_service_image" {
+  description = "Container image for JVS API service."
+  type        = string
+  default     = "gcr.io/cloudrun/hello:latest"
+}
+
+variable "jvs_api_service_iam" {
+  description = "IAM member bindings for JVS API service."
+  type = object({
+    admins     = list(string)
+    developers = list(string)
+    invokers   = list(string)
+  })
+  default = {
+    admins     = []
+    developers = []
+    invokers   = []
+  }
+}
```

Comment:
I see this pattern repeated multiple times (service name, service account, service image, service iam), and it feels a little strange. Is there a reason these need to be variables?

Reply:
At least half of them are required:

- Service accounts are created in a separate module so we need to pass them over
- Images are built separately so we need to pass them over

The service IAM and name are optional. For simplicity, I can get rid of them for now.

Reply:
Removed IAM and service name

Reply:
Ack. Tracking in https://github.com/abcxyz/jvs/issues/196